### PR TITLE
add atan to algorithms::math

### PR DESCRIPTION
**Pull request by PrometheusPi**
This pull request adds `atan` to `algorithm::math::` in **libPMacc** as requested by @psychocoderHPC in #1468.

### CODE REVIEW

**Comment by PrometheusPi** [comment]
@psychocoderHPC Since you are reviewing #1468, I assigned you for the review of this pull request as well.

File: src/libPMacc/include/algorithms/math/floatMath/trigo.tpp [hunk 1]
```diff
@@ -99,6 +99,17 @@ struct Tan<float>
 };
 
 template<>
+struct ATan<float>
+{
+    typedef float result;
+
+    HDINLINE float operator( )(const float& value )
```
Comment (psychocoderHPC) [inline src/libPMacc/include/algorithms/math/floatMath/trigo.tpp]: space before  `)`

File: src/libPMacc/include/algorithms/math/doubleMath/trigo.tpp [hunk 2]
```diff
@@ -90,6 +90,17 @@ struct Tan<double>
 };
 
 template<>
+struct ATan<double>
+{
+    typedef double result;
+
+    HDINLINE double operator( )(const double& value )
```
Comment (psychocoderHPC) [inline src/libPMacc/include/algorithms/math/doubleMath/trigo.tpp]: space before `)`

**Comment by PrometheusPi** [comment]
@psychocoderHPC fixed

**Comment by PrometheusPi** [comment]
@psychocoderHPC thanks. I will rebase #1468.

**Pull request merged**